src/lidar_subscriber.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <memory>
+#include <cmath>
+#include "rclcpp/rclcpp.hpp"
+
+#include "core-structs.h"
+
+
+#include "sensor_msgs/msg/laser_scan.hpp"
+#include <queue>
+#include <limits>
+#include "visualization_msgs/msg/marker.hpp"
+
+constexpr float LANDMARK_THRESHOLD_m = 0.02;
+constexpr bool testing = true;
+
+bool same_landmark(const Observation2D &first, const Observation2D &second){
```

Comment:
This function should be static since it's only contained in this file.

Reply:
Done!